src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -229,11 +225,18 @@ protected ConnectionState readHTTPInitial(HttpRequest httpRequest) {
      */
     private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         // Make a copy of the original request
-        this.currentRequest = copy(httpRequest);
+        final HttpRequest currentRequest = copy(httpRequest);
 
         // Set up our filters based on the original request. If the HttpFiltersSource returns null (meaning the request/response
         // should not be filtered), fall back to the default no-op filter source.
-        HttpFilters filterInstance = proxyServer.getFiltersSource().filterRequest(currentRequest, ctx);
+        HttpFilters filterInstance;
+        try {
+            filterInstance = proxyServer.getFiltersSource().filterRequest(currentRequest, ctx);
+        } finally {
+            if (currentRequest instanceof ReferenceCounted) {
+                ((ReferenceCounted)currentRequest).release();
+            }
+        }
```

Comment:
I think this copy is here so original request can not be mutated in request filter written by a user (by VGS team for example)